pkg/storage/stores/series/series_index_store.go

```diff
@@ -244,31 +243,57 @@ func (c *indexReaderWriter) chunksToSeries(ctx context.Context, in []logproto.Ch
 		}
 	}
 
-	for _, group := range groups {
-		sort.Sort(group)
-		chunks, err := c.fetcher.FetchChunks(ctx, group.chunks, group.keys)
-		if err != nil {
-			return nil, err
-		}
+	type f func() ([]labels.Labels, error)
+	var jobs []f
 
-	outer:
-		for _, chk := range chunks {
-			for _, matcher := range matchers {
-				if matcher.Name == astmapper.ShardLabel || matcher.Name == labels.MetricName {
-					continue
+	for _, g := range groups {
+    group := g
+		jobs = append(jobs, f(func() ([]labels.Labels, error) {
+			sort.Sort(group)
+			chunks, err := c.fetcher.FetchChunks(ctx, group.chunks, group.keys)
+			if err != nil {
+				return nil, err
+			}
+
+		results := make([]labels.Labels, 0, len(chunks))
```

Comment:
you're defining results here and at line 281

Reply:
it might help to rename one for clarity nevertheless, will do!